Basic/APIDiorama/Assets/NetvarVsRpc/Scripts/ColorManager.cs

```diff
@@ -0,0 +1,93 @@
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Manages the color of a Networked object
+    /// </summary>
+    public class ColorManager : NetworkBehaviour
+    {
+        [SerializeField]
+        bool m_UseNetworkVariableForColor;
+
+        NetworkVariable<Color32> m_NetworkedColor = new NetworkVariable<Color32>();
+        Material m_Material;
+
+        void Awake()
+        {
+            m_Material = GetComponent<Renderer>().material;
+        }
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            if (IsClient)
+            {
+                if (m_UseNetworkVariableForColor)
+                {
+                    OnClientColorChanged(m_Material.color, m_NetworkedColor.Value);
+                    m_NetworkedColor.OnValueChanged += OnClientColorChanged;
+                }
+            }
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            base.OnNetworkDespawn();
+            if (IsClient)
+            {
+                if (m_UseNetworkVariableForColor)
+                {
+                    m_NetworkedColor.OnValueChanged -= OnClientColorChanged;
+                }
+            }
+        }
+
+        void Update()
+        {
+            if (!IsClient)
+            {
+                return;
+            }
+
+            bool playerIsCloseEnough = true; //todo: implement behaviour
+            if (playerIsCloseEnough)
```

Comment:
About this one: I'll either impklement or remove it depending on how we decide to manage interactions

Reply:
To keep things as atomic as possible, I'm removing this check from